src/NuGet.Clients/NuGet.VisualStudio.Implementation/Extensibility/VsPackageInstaller.cs

```diff
@@ -271,10 +270,14 @@ private Action<string> ErrorHandler
             {
                 return msg =>
                     {
-                        if (_projectContext != null)
+                        // instead of initializing it in MEF constructor since it access ISourceControlManagerProvider
+                        // initializing it here so that it doesn't hang at MEF initialization.
+                        if (ProjectContext == null)
```

Comment:
That's not obvious, as I looked at its calling tree it's quite wide and deep. Also the fact it runs under JTF doesn't guarantee the thread-safety. For instance in one of external methods it can switch to the thread pool or spawn multiple tasks.